library/L1_Peripheral/pin.hpp

```diff
@@ -93,26 +98,29 @@ class Pin
   /// the job of the L1 peripheral that uses this pin to manage its own function
   /// code usage.
   ///
-  /// @param function - pin function code
+  /// @param function pin function code
   virtual void SetPinFunction(uint8_t function) const = 0;
+
   /// Set pin's resistor pull, setting ot either no resistor pull, pull down,
   /// pull up and repeater.
   ///
   /// @param resistor - which resistor setup you would like.
   virtual void SetPull(Resistor resistor) const = 0;
```

Comment:
Couldn't this return an error if the resistor state desired does not exist for that platform?